cli/cmd/plugin/unmanaged-cluster/cluster/kind.go

```diff
@@ -158,6 +163,64 @@ func (kcm KindClusterManager) Prepare(c *config.UnmanagedClusterConfig) error {
 	return nil
 }
 
+// PreflightCheck performs any pre-checks that can find issues up front that
+// would cause problems for cluster creation.
+func (kcm KindClusterManager) PreflightCheck() []error {
+	// Check presence of docker
+	cmd := exec.Command("docker", "ps")
+	if err := cmd.Run(); err != nil {
+		// In this case we can't check the rest of the settings, so just return
+		// the one error.
+		return []error{errors.New("docker CLI not found or invalid permissions")}
```

Comment:
This could fail for additional reasons, such as the docker daemon/engine not being installed. I think we may need a more generic message. Something like:

```
docker is not installed or not reachable. Verify it's installed, running, and your user has permissions to interact with it. Error when attempting to run docker ps: %w
```

Additionally, I think it'd be of value to return whatever `docker ps` returned.

Reply:
Done - with one caveat. On failure there is no output, but the error returned has what we need and will be formatted into the output. So we can stick with `cmd.Run()` here and get output similar to:

`Error when attempting to run docker ps: exec: "docker": executable file not found in $PATH`